modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -75,25 +117,66 @@ package client {
     * Interpreters
     */
   object Giapi {
+    private final case class StatusStreamer(aggregate: StatusHandlerAggregate, ss: StatusService)
+
+    private def statusGetter[F[_]: Sync](c: ActiveMQJmsProvider): F[StatusGetter] = Sync[F].delay {
+      val sg = new StatusGetter("client")
+      sg.startJms(c)
+      sg
+    }
+
+    private def commandSenderClient[F[_]: Applicative](c: ActiveMQJmsProvider): F[CommandSenderClient] =
+      Applicative[F].pure { new CommandSenderClient(c) }
+
+    private def statusStreamer[F[_]: Sync](c: ActiveMQJmsProvider): F[StatusStreamer] = Sync[F].delay {
+      val aggregate     = new StatusHandlerAggregate()
+      val statusService = new StatusService(aggregate, "statusService", "*")
+      statusService.startJms(c)
+      StatusStreamer(aggregate, statusService)
+    }
+
+    private def streamItem[F[_]: Effect, A: ItemGetter](agg: StatusHandlerAggregate, statusItem: String,
+                                       ec: ExecutionContext): F[Stream[F, A]] =
+      Sync[F].delay {
+        implicit val exc: ExecutionContext = ec
+        def statusHandler(q: async.mutable.Queue[F, A]) = new StatusHandler {
+
+          override def update[B](item: StatusItem[B]): Unit =
+          // Check the item name and attempt convert it to A
+            if (item.getName === statusItem) {
+              ItemGetter[A].value(item.getValue).foreach { a =>
+                async.unsafeRunAsync(q.enqueue1(a))(_ => IO.unit)
```

Comment:
I am always suspicious about the use of `unsafeRunAsync` in the middle of a computation. The rule of thumb is to have only one call to `unsafeRunAsync` at the end of the world. 
The usual case is that you somehow are doing something like `IO(async.unsafeRunAsync(IO(something))`, that could be just `IO(something)`. But, in this code I don't see how to get rid of it.

Reply:
I don't think you have any choice here because the API requires you to side-effect.